blocks/dfio.py

```diff
@@ -75,71 +60,49 @@ def write_df(df, datafile, **write_args):
     Check the pandas doc for more information on the supported arguments
-
     """
-    extension, compression = _get_extension(datafile.path)
+    extension = _get_extension(path)
     write_name = _writers[extension]
-    # infer compression from filepath or from explicit arg
-    compression = compression or write_args.get("compression")
-    with datafile.handle("wb") as buf:
-        # Some customizations for different file types
-        if write_name == "to_avro":
-            return _write_avro(df, buf, **write_args)
-
-        if (
-            write_name == "to_parquet"
-            and not pd.Series(df.columns).map(type).eq(str).all()
-        ):
-            warnings.warn(
-                "Dataframe contains non-string column names, which cannot be saved in parquet.\n"
-                "Blocks will attempt to convert them to strings."
-            )
-            df.columns = df.columns.astype("str")
-
-        if write_name == "to_json":
-            defaults = {"lines": True, "orient": "records"}
-            defaults.update(write_args)
-            write_args = defaults
-
-        if write_name == "to_csv":
-            # make index=False the default for similar behaviour to other formats
-            write_args["index"] = write_args.get("index", False)
-
-        # For csv and pickle we have to manually compress
-        manual_compress = write_name in ("to_csv", "to_pickle", "to_json")
-        if manual_compress:
-            write_args[
-                "compression"
-            ] = None  # default "infer" incompatible with handles as of 0.24
-
-        if manual_compress and compression is not None:
-            raise ValueError(
-                "Compression {} is not supported for CSV/Pickle/JSON".format(
-                    compression
-                )
-            )
 
-        if write_name in ("to_csv", "to_json"):
-            buf = TextIOWrapper(buf, write_through=True)
+    # Some customizations for different file types
+    if write_name == "to_avro":
+        return _write_avro(df, path, **write_args)
 
-        write_fn = getattr(df, write_name)
-        write_fn(buf, **write_args)
+    if write_name == "to_parquet" and not pd.Series(df.columns).map(type).eq(str).all():
+        warnings.warn(
+            "Dataframe contains non-string column names, which cannot be saved in parquet.\n"
+            "Blocks will attempt to convert them to strings."
+        )
+        df.columns = df.columns.astype("str")
+
+    if write_name == "to_json":
+        defaults = {"lines": True, "orient": "records"}
+        defaults.update(write_args)
+        write_args = defaults
 
+    if write_name == "to_csv":
+        # make index=False the default for similar behaviour to other formats
+        write_args["index"] = write_args.get("index", False)
 
-def _read_avro(handle, **read_args):
+    write_fn = getattr(df, write_name)
+    write_fn(path, **write_args)
+
+
+def _read_avro(path, **read_args):
```

Comment:
nit: are you passing the **read args?

Reply:
in this case no additional read args are implemented, but left them here for consistency and if we add any in the future